src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -1661,6 +1662,17 @@ def add_dns_srv_record(resource_group_name, zone_name, record_set_name, priority
 def add_dns_txt_record(resource_group_name, zone_name, record_set_name, value):
     record = TxtRecord(value)
     record_type = 'txt'
+    long_text = ''.join(x for x in record.value)
+    long_text = long_text.replace('\\', '')
+    original_len = len(long_text)
+    record.value = []
+    while len(long_text) > 255:
+        record.value.append(long_text[:255])
+        long_text = long_text[255:]
+    record.value.append(long_text)
+    final_str = ''.join(record.value)
+    final_len = len(final_str)
+    assert original_len == final_len
```

Comment:
This is like a 'this should never happen' check. If for whatever reason it does happen, do you want to block the record being saved?
I guess you do want to block saving so maybe `assert original_len == final_len, "Unable to create txt record"`.
Consider using CLI error as well.